### PR TITLE
enhancement: handle Ledger EVM transactions when blind signing is disabled

### LINKED ISSUE #432: ️🛠 - Handle Ledger EVM transactions when blind signing disabled
### Task description

We do not account for when the user has blind signing disabled **for the Ethereum app**. We need to make sure the step-by-step guide is applicable for both apps.

### Requirements

- [x] Add method of getting the app settings
- [x] Add UI/logic for when trying to sign but is disabled
- [x] Fix blind signing popup to work with either SMR/EVM
- [x] Fix logic so that request waits for user to enable blind signing or just make it work somehow
- [x] Fix closing blind signing not stopping the transfer in the background
- [x] Make L1 blind signing tx closable (investigate this flow in general)
- [x] Renaming the default poll constant to use device state instead of status

### PR DESCRIPTION
**Pull request by maxwellmattryan**
## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

...

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [x] Linux
    -   [ ] Windows

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [x] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/desktop/views/dashboard/send-flow/views/TransactionSummaryView.svelte [hunk 1]
```diff
@@ -51,7 +51,7 @@
     async function onConfirmClick(): Promise<void> {
         try {
             if (isAssetFromLayer2) {
-                await sendTransactionFromEvm(preparedTransaction, chain, closePopup)
+                await sendTransactionFromEvm(preparedTransaction, chain, () => {})
```
Comment (MarkNerdi) [inline packages/desktop/views/dashboard/send-flow/views/TransactionSummaryView.svelte]: if we dont need a callback anymore, remove that param

File: packages/desktop/lib/electron/utils/ledger.utils.ts [hunk 2]
```diff
@@ -1,6 +1,9 @@
 import AppEth from '@ledgerhq/hw-app-eth'
 import TransportNodeHid from '@ledgerhq/hw-transport-node-hid'
 import { listen } from '@ledgerhq/logs'
+import { IEvmTransactionSignature } from '@core/layer-2/interfaces'
```
Comment (nicole-obrien) [inline packages/desktop/lib/electron/utils/ledger.utils.ts]: duplicate imports

File: packages/desktop/lib/electron/utils/ledger.utils.ts [hunk 3]
```diff
@@ -24,6 +27,18 @@ export async function closeTransport(): Promise<void> {
     }
 }
 
+export async function getEthereumAppSettings(): Promise<ILedgerEthereumAppSettings> {
+    const appEth = new AppEth(transport)
+    const settings = await appEth.getAppConfiguration()
+    return {
+        version: settings.version,
+        arbitraryDataEnabled: Boolean(settings.arbitraryDataEnabled),
```
Comment (nicole-obrien) [inline packages/desktop/lib/electron/utils/ledger.utils.ts]: type error

File: packages/desktop/views/dashboard/Dashboard.svelte [hunk 4]
```diff
@@ -63,7 +63,7 @@
     onDestroy(() => {
         Platform.DeepLinkManager.clearDeepLinkRequest()
         if ($isActiveLedgerProfile) {
-            stopPollingLedgerNanoStatus()
+            stopPollingLedgerDeviceState()
```
Comment (nicole-obrien) [inline packages/desktop/views/dashboard/Dashboard.svelte]: should this be in logout?

File: packages/shared/src/lib/core/ledger/helpers/deconstructLedgerDeviceStatePollingConfiguration.ts [hunk 5]
```diff
@@ -1,11 +1,11 @@
 import { profileManager as _profileManager } from '@core/profile-manager/stores'
 
 import { DEFAULT_LEDGER_NANO_STATUS_POLL_INTERVAL } from '../constants'
-import { ILedgerNanoStatusPollingConfiguration } from '../interfaces'
+import { ILedgerDeviceStatePollingConfiguration } from '../interfaces'
 
-export function deconstructLedgerNanoStatusPollingConfiguration(
-    config: ILedgerNanoStatusPollingConfiguration
-): ILedgerNanoStatusPollingConfiguration {
+export function deconstructLedgerDeviceStatePollingConfiguration(
```
Comment (nicole-obrien) [inline packages/shared/src/lib/core/ledger/helpers/deconstructLedgerDeviceStatePollingConfiguration.ts]: is this function overkill

File: packages/shared/src/lib/core/ledger/interfaces/ledger-app-settings.interface.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { LedgerAppName } from '../enums'
+
+export interface ILedgerAppSettings {
+    [LedgerAppName.Shimmer]?: ILedgerAppBaseSettings
+    [LedgerAppName.Ethereum]?: ILedgerEthereumAppSettings
+}
+
+export interface ILedgerEthereumAppSettings extends ILedgerAppBaseSettings {
```
Comment (nicole-obrien) [inline packages/shared/src/lib/core/ledger/interfaces/ledger-app-settings.interface.ts]: seperate file?

File: packages/shared/src/lib/core/ledger/interfaces/ledger-app-settings.interface.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    erc20ProvisioningNecessary: boolean
+    starkEnabled: boolean
+    starkv2Supported: boolean
+}
+
+interface ILedgerAppBaseSettings {
```
Comment (nicole-obrien) [inline packages/shared/src/lib/core/ledger/interfaces/ledger-app-settings.interface.ts]: seperate file?

File: packages/shared/src/lib/core/ledger/utils/buildLedgerDeviceState.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    status: LedgerNanoStatus,
+    ethereumAppSettings?: ILedgerEthereumAppSettings
+): ILedgerDeviceState {
+    return <ILedgerDeviceState>{
+        connected: ethereumAppSettings ? true : status.connected,
+        locked: ethereumAppSettings ? false : status.locked,
+        device: status.device,
+        app: status.app?.name as LedgerAppName,
+        settings: <ILedgerAppSettings>{
+            ...(status.app?.name === LedgerAppName.Shimmer && {
```
Comment (nicole-obrien) [inline packages/shared/src/lib/core/ledger/utils/buildLedgerDeviceState.ts]: type error

File: packages/shared/src/lib/core/profile-manager/actions/events-handlers/handleTransactionProgressEvent.ts [hunk 9]
```diff
@@ -68,6 +72,21 @@ function openPopupIfVerificationNeeded(progress: TransactionProgress): void {
                     id: PopupId.EnableLedgerBlindSigning,
                     hideClose: true,
```
Comment (nicole-obrien) [inline packages/shared/src/lib/core/profile-manager/actions/events-handlers/handleTransactionProgressEvent.ts:74]: should be able to close it

**Review by nicole-obrien — CHANGES_REQUESTED** [review]

File: packages/desktop/views/dashboard/Dashboard.svelte (hunk 4 above)
Comment (maxwellmattryan) [reply packages/desktop/views/dashboard/Dashboard.svelte]: Seems like it's already there too

File: packages/shared/src/lib/core/profile-manager/actions/events-handlers/handleTransactionProgressEvent.ts (hunk 9 above)
Comment (maxwellmattryan) [reply packages/shared/src/lib/core/profile-manager/actions/events-handlers/handleTransactionProgressEvent.ts:74]: This has to do with the L1 flow, maybe we should test that too before making this change. Will add for L2 though :+1:

**Review by nicole-obrien — CHANGES_REQUESTED** [review]
Seems to work on shimmer evm now, but errors on shimmer.
- crashes when trying to send without blind signing

**Review by nicole-obrien — APPROVED** [review]

**Pull request merged**